utilities/inference_utils.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+import json
+import re
+import shlex
+from functools import cache
+from json import JSONDecodeError
+from string import Template
+from typing import Any, Dict, Optional
+from urllib.parse import urlparse
+
+from ocp_resources.inference_service import InferenceService
+from pyhelper_utils.shell import run_command
+from simple_logger.logger import get_logger
+
+from utilities.manifests.runtime_query_config import RUNTIMES_QUERY_CONFIG
+
+LOGGER = get_logger(name=__name__)
+INFERENCE_QUERIES: Dict[str, Dict[str, str]] = {
+    "nitrogen-boil-temp": {
+        "text": "At what temperature does liquid Nitrogen boil?",
+        "response_text": "74 degrees F",
+    }
+}
+
+
+class Inference:
+    ALL_TOKENS: str = "all-tokens"
+    STREAMING: str = "streaming"
+
+    def __init__(self, inference_service: InferenceService, runtime: str, protocol: str, inference_type: str):
+        """
+        Args:
+            inference_service: InferenceService object
+        """
+        self.inference_service = inference_service
+        self.runtime = runtime
+        self.protocol = protocol
+        self.inference_type = inference_type
+
+    @cache
+    def get_inference_url(self) -> str:
+        if url := self.inference_service.instance.status.components.predictor.url:
+            return urlparse(url).netloc
+        else:
+            raise ValueError(f"{self.inference_service.name}: No url found in InferenceService status")
+
+    @cache
+    def get_inference_config(self) -> Dict[str, Any]:
+        if runtime_config := RUNTIMES_QUERY_CONFIG.get(self.runtime):
+            if inference_type := runtime_config.get(self.inference_type):
+                if data := inference_type.get(self.protocol):
+                    return data
+
+                else:
+                    raise ValueError(
+                        f"Protocol {self.protocol} not supported.\nSupported protocols are {inference_type}"
+                    )
+
+            else:
+                raise ValueError(
+                    f"Inference type {inference_type} not supported.\n"
+                    f"Supported inference types are {runtime_config['endpoints']}"
```

Comment:
What are these endpoints and why are you using them to specify what "inference types" are supported?